lms/validation/authentication/_oauth.py

```diff
@@ -93,27 +94,26 @@ def state_param(self):
 
         return jwt_str
 
-    def lti_user(self) -> LTIUser:
+    def lti_user(self, state=None) -> LTIUser:
```

Comment:
This was needed at one point because the Canvas Studio auth flow needed to read the `state` value out of a cookie. However since then I changed the `/api/canvas_studio/oauth/callback` endpoint so that if called without a `state` arg, it will read the `state` arg and then redirect to a URL with it added. Hence this change is not strictly required any more. It might turn out to be useful again in future. I can remove it from this PR though.